olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -624,6 +626,75 @@ PartitionsRepository::QueryPartitionsInBatches(
                                               aggregated_network_statistics);
 }
 
+client::ApiNoResponse PartitionsRepository::ParsePartitionsStream(
+    std::shared_ptr<AsyncJsonStream> async_stream,
+    PartitionsStreamCallback partition_callback,
+    client::CancellationContext context) {
+  rapidjson::ParseResult parse_result;
+
+  // Retry to parse the stream until it's closed.
+  while (!async_stream->IsClosed()) {
```

Comment:
Can we limit the number of attempts?

Reply:
It's limited by retry count in other task, once the download task is finished, the stream is closed